mngapi/barong/client_test.go

```diff
@@ -53,3 +54,87 @@ func TestCreateNewClient(t *testing.T) {
 		assert.EqualError(t, err, "Invalid Key: Key must be PEM encoded PKCS1 or PKCS8 private key")
 	})
 }
+
+func TestCreateServiceAccount(t *testing.T) {
+	t.Run("Success response", func(t *testing.T) {
+		client, _ := New(URL, jwtIssuer, jwtAlgo, jwtPrivateKey)
+
+		expected := `{"email":"test+SI0388B7681C@yellow.com","uid":"SI0388B7681C","role":"service_account","level":3,"state":"active","user":{"email":"test@test.com","uid":"IDCA2AC08296","role":"superadmin","level":3,"otp":true,"state":"active","referral_uid":"","data":"{\"onboarding\":true,\"language\":\"en\"}"},"created_at":"2021-02-15T10:15:18Z","updated_at":"2021-02-15T10:15:18Z"}`
+		client.mngapiClient = &MockClient{
+			response: []byte(expected),
+			apiError: nil,
+		}
+
+		params := CreateServiceAccountParams{
+			OwnerUID: "IDCA2AC08296",
+			Role:     "service_account",
+		}
+		serviceAccount, err := client.CreateServiceAccount(params)
+
+		result, _ := json.Marshal(serviceAccount)
+		assert.Nil(t, err)
+		assert.Equal(t, result, []byte(expected))
+	})
+
+	t.Run("Error user doesn't exist", func(t *testing.T) {
+		client, _ := New(URL, jwtIssuer, jwtAlgo, jwtPrivateKey)
+
+		client.mngapiClient = &MockClient{
+			response: nil,
+			apiError: &mngapi.APIError{
+				StatusCode: 422,
+				Error:      "User doesnt exist",
+			},
+		}
+
+		params := CreateServiceAccountParams{
+			OwnerUID: "ID123456789",
+			Role:     "service_account",
+		}
+		serviceAccount, err := client.CreateServiceAccount(params)
+
+		assert.NotNil(t, err)
+		assert.Equal(t, err.StatusCode, 422)
+		assert.Equal(t, err.Error, "User doesnt exist")
+		assert.Nil(t, serviceAccount)
+	})
+}
+
+func TestDeleteServiceAccountByUID(t *testing.T) {
+	t.Run("Success response", func(t *testing.T) {
+		client, _ := New(URL, jwtIssuer, jwtAlgo, jwtPrivateKey)
```

Comment:
also here in test it always good to retrive the error and assert:
```assert.NoError(t, err)```

Reply:
Fixed.